servant/src/Servant/API/ContentTypes.hs

```diff
@@ -219,9 +226,10 @@ instance AllMime '[] where
     allMime _ = []
 
 instance (Accept ctyp, AllMime ctyps) => AllMime (ctyp ': ctyps) where
-    allMime _ = (contentType pctyp):allMime pctyps
-      where pctyp  = Proxy :: Proxy ctyp
-            pctyps = Proxy :: Proxy ctyps
+    allMime _ = NE.toList (contentTypes pctyp) ++ allMime pctyps
+      where
+        pctyp  = Proxy :: Proxy ctyp
+        pctyps = Proxy :: Proxy ctyps
```

Comment:
Well, tbh, I wish we had less cryptic type level variables, too. I would strongly prefer `contentTypes` over `ctyps`, both at the term and the type level.